frontend/src/container/PublicProfile.jsx

```diff
@@ -62,46 +81,83 @@ export class PublicProfile extends React.Component {
     }]))
   }
 
+  handleFetchErrors = (result, errorList, defaultErrorMessage) => {
+    if (result.status < 400) return false
+
+    const { props } = this
+    const code = result.json.code
+    const error = errorList.find(m => m.status === result.status && m.code === code) || { message: defaultErrorMessage }
+    props.dispatch(newFlashMessage(error.message))
+    props.history.push(PAGE.HOME)
+    return true
+  }
+
   getUser = async () => {
     const { props } = this
     const userId = props.match.params.userid
 
     const fetchGetUser = await props.dispatch(getAboutUser(userId))
+    if (this.handleFetchErrors(
+      fetchGetUser,
+      [{ status: 400, code: 1001, message: props.t('Unknown user') }],
+      props.t('Error while loading user')
+    )) return
+
+    const apiUser = { ...serialize(fetchGetUser.json, serializeUserProps) }
+    this.setState(oldState => {
```

Comment:
the convention is to name the parameter `prevState` or `previousState`

Reply:
If it's ok with you I'll address this naming issue with #4088.

Reply:
sure